src/DIRAC/FrameworkSystem/Service/BundleDeliveryHandler.py

```diff
@@ -1,66 +1,66 @@
 """ Handler for CAs + CRLs bundles
 """
-import tarfile
-import os
 import io
+import os
+import tarfile
 
+from DIRAC import S_ERROR, S_OK, gConfig, gLogger
 from DIRAC.Core.DISET.RequestHandler import RequestHandler
-from DIRAC import gLogger, S_OK, S_ERROR, gConfig
-from DIRAC.Core.Utilities.ThreadScheduler import gThreadScheduler
-from DIRAC.Core.Utilities import File, List
 from DIRAC.Core.Security import Locations, Utilities
+from DIRAC.Core.Utilities import File, List
 
 
 class BundleManager:
+    """Utility class"""
+
     def __init__(self, baseCSPath):
         self.__csPath = baseCSPath
         self.__bundles = {}
         self.updateBundles()
 
     def __getDirsToBundle(self):
         dirsToBundle = {}
-        result = gConfig.getOptionsDict("%s/DirsToBundle" % self.__csPath)
+        result = gConfig.getOptionsDict(f"{self.__csPath}/DirsToBundle")
         if result["OK"]:
             dB = result["Value"]
             for bId in dB:
                 dirsToBundle[bId] = List.fromChar(dB[bId])
-        if gConfig.getValue("%s/BundleCAs" % self.__csPath, True):
+        if gConfig.getValue(f"{self.__csPath}/BundleCAs", True):
             dirsToBundle["CAs"] = [
-                "%s/*.0" % Locations.getCAsLocation(),
-                "%s/*.signing_policy" % Locations.getCAsLocation(),
-                "%s/*.pem" % Locations.getCAsLocation(),
+                os.path.join(Locations.getCAsLocation()),
+                "*.0",
+                os.path.join(Locations.getCAsLocation()),
+                "*.signing_policy",
+                os.path.join(Locations.getCAsLocation()),
+                "*.pem",
             ]
-        if gConfig.getValue("%s/BundleCRLs" % self.__csPath, True):
-            dirsToBundle["CRLs"] = ["%s/*.r0" % Locations.getCAsLocation()]
+        if gConfig.getValue(f"{self.__csPath}/BundleCRLs", True):
+            dirsToBundle["CRLs"] = [os.path.join(Locations.getCAsLocation(), "*.r0")]
         return dirsToBundle
 
-    def getBundles(self):
-        return {bId: self.__bundles[bId] for bId in self.__bundles}
-
     def bundleExists(self, bId):
         return bId in self.__bundles
 
     def getBundleVersion(self, bId):
         try:
             return self.__bundles[bId][0]
-        except Exception:
+        except KeyError:
```

Comment:
https://github.com/DIRACGrid/DIRAC/blob/b59c427108485097be3ddf1c33b5c4cb7091bc7c/src/DIRAC/FrameworkSystem/Service/BundleDeliveryHandler.py#L80
These are tuples, so it would throw `IndexError`